ci/scripts/cpp_test.sh

```diff
@@ -32,13 +32,36 @@ export LD_LIBRARY_PATH=${ARROW_HOME}/${CMAKE_INSTALL_LIBDIR:-lib}:${LD_LIBRARY_P
 # to retrieve metadata. Disable this so that S3FileSystem tests run faster.
 export AWS_EC2_METADATA_DISABLED=TRUE
 
+ctest_options=()
 case "$(uname)" in
   Linux)
     n_jobs=$(nproc)
     ;;
   Darwin)
     n_jobs=$(sysctl -n hw.ncpu)
     ;;
+  MINGW*)
+    n_jobs=${NUMBER_OF_PROCESSORS:-1}
+    exclude_tests="gandiva-internals-test"
```

Comment:
I assume we'll need a follow-up jira to fix these tests.

Reply:
Done: https://issues.apache.org/jira/browse/ARROW-9072

Reply:
Thanks!